File: Team6220_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team6220_CENTERSTAGE/Localizer.java
package org.firstinspires.ftc.team6220_CENTERSTAGE;

import com.acmerobotics.roadrunner.Time;
import com.acmerobotics.roadrunner.Twist2dDual;
/*
This is part of roadrunner.
 */
public interface Localizer {
    Twist2dDual<Time> update();
}

File: RoadRunner/src/main/kotlin/com/acmerobotics/roadrunner/localization/Localizer.kt
package com.acmerobotics.roadrunner.localization

import com.acmerobotics.roadrunner.geometry.Pose2d

/**
 * Generic interface for estimating robot pose over time.
 */
interface Localizer {

    /**
     * Current robot pose estimate.
     */
    var poseEstimate: Pose2d

    /**
     * Current robot pose velocity (optional)
     */
    val poseVelocity: Pose2d?

    /**
     * Completes a single localization update.
     */
    fun update()
}
